src/Widgets/PasswordEditor.vala

```diff
@@ -183,24 +181,23 @@ namespace SwitchboardPlugUserAccounts.Widgets {
                     pw_level.set_tooltip_text (_("Strong password strength"));
                 }
 
+                if (quality >= 0) {
+                    is_obscure = true;
+                    error_new_revealer.reveal_child = false;
+                } else {
+                    var pw_error = (PasswordQuality.Error) quality;
+                    var error_string = pw_error.to_string (error);
+
+                    error_new_label.label = "<span font_size=\"small\">%s</span>".printf (error_string);
+                    error_new_revealer.reveal_child = true;
+                    is_obscure = false;
+                }
+
                 /*
                  * without superuser privileges your new password needs to pass an obscurity test
                  * which is based on passwd's one to guess passwd's response.
                  */
-                if (!is_authenticated) {
-                    if (quality >= 0) {
-                        is_obscure = true;
-                        error_new_revealer.set_reveal_child (false);
-                    } else {
-                        var pw_error = (PasswordQuality.Error) quality;
-                        var error_string = pw_error.to_string (error);
-
-                        error_new_label.label = "<span font_size=\"small\">%s</span>".printf (error_string);
-                        error_new_revealer.set_reveal_child (true);
-                        is_obscure = false;
-                    }
-                } else {
-                    //a superuser does not need to care about obscurity
+                if (is_authenticated) {
```

Comment:
Can we move this condition to the above `} else {` part? Currently we could assign `is_obscure` twice to the same value. The best solution to simplify this would be to replace `is_obscure = false;` in the diff line 193 to `is_obscure = is_authenticated`.